mentoring/mentoring.py

```diff
@@ -25,70 +25,226 @@
 
 import logging
 
+from collections import namedtuple
+
 from xblock.core import XBlock
-from xblock.fields import Boolean, Scope, String
+from xblock.exceptions import NoSuchViewError
+from xblock.fields import Boolean, Scope, String, Integer, Float, List
+from xblock.fragment import Fragment
+from xblock.validation import ValidationMessage
 
-from .light_children import XBlockWithLightChildren
-from .message import MentoringMessageBlock
-from .utils import get_scenarios_from_path, load_resource, render_template
+from components import MentoringMessageBlock
+from components.step import StepParentMixin, StepMixin
 
+from xblockutils.resources import ResourceLoader
+from xblockutils.studio_editable import StudioEditableXBlockMixin, StudioContainerXBlockMixin
 
 # Globals ###########################################################
 
 log = logging.getLogger(__name__)
-
+loader = ResourceLoader(__name__)
 
 # Classes ###########################################################
 
-class MentoringBlock(XBlockWithLightChildren):
+Score = namedtuple("Score", ["raw", "percentage", "correct", "incorrect", "partially_correct"])
+
+
+@XBlock.needs("i18n")
+class MentoringBlock(XBlock, StepParentMixin, StudioEditableXBlockMixin, StudioContainerXBlockMixin):
     """
     An XBlock providing mentoring capabilities
 
-    Composed of text, answers input fields, and a set of multiple choice quizzes with advices. 
-    A set of conditions on the provided answers and quizzes choices will determine if the
+    Composed of text, answers input fields, and a set of MRQ/MCQ with advices.
+    A set of conditions on the provided answers and MCQ/MRQ choices will determine if the
     student is a) provided mentoring advices and asked to alter his answer, or b) is given the
     ok to continue.
     """
-    attempted = Boolean(help="Has the student attempted this mentoring step?",
-                        default=False, scope=Scope.user_state)
-    completed = Boolean(help="Has the student completed this mentoring step?",
-                        default=False, scope=Scope.user_state)
-    next_step = String(help="url_name of the next step the student must complete (global to all blocks)",
-                       default='mentoring_first', scope=Scope.preferences)
-    followed_by = String(help="url_name of the step after the current mentoring block in workflow",
-                         default=None, scope=Scope.content)
-    url_name = String(help="Name of the current step, used for URL building",
-                      default='mentoring', scope=Scope.content)
```

Comment:
If you replace url_name by a generated value, is the url_name value going to be the same in existing XBlocks? If not this will break existing `jump_to_id` links in the courseware content.